### ISSUE TITLE
loss nan

### ISSUE THREAD

**Issue by yarkable** [CASIA-IVA-Lab/TDKD#4]
Hi, thanks for your great work. But I just cloned this repo and run it to train a distillation model. The loss became nan after nearly 20 epochs. The teacher and student models are loaded properly. 

By the way, are the results reported in the paper trained with a pretrained student other than a student trained from scratch? Looking forward to your reply.

Sincerely.

**Comment by zxDeepDiver** [comment]
Thank you for paying attention to our work. Yes，the student model should be normally trained to get a good initialization for the distillation. How many GPUs  did you use? We use 2 GPUs to train the model.

**Comment by zxDeepDiver** [comment]
Thank you for paying attention to our work. Yes，the student model should be normally trained to get a good initialization for the distillation.  Are the the widerface-pretrained student weights loaded properly?

**Comment by yarkable** [comment]
Yep, I see that in your code, A `base.pth` is loaded as default. And I just load it as the code does. I also eval the `base.pth`, and it get scores of 87, 85 and 79 in widerface.

BTW, during the training process, I notice that the `Dfeat loss` is 50-60 while location loss and classification loss is just 1-3, is there anything wrong with it or it's just normal? 

Best.

**Comment by zxDeepDiver** [comment]
@icekrn

**Comment by yarkable** [comment]
Hi, I've tried for a lot of time. Single training student or teacher yields a good performance. But in distillation mode, the loss always gets Nan. So could you please rerun the code?

**Comment by zxDeepDiver** [comment]
We are checking that

**Comment by zxDeepDiver** [comment]
We are checking that

**Comment by yarkable** [comment]
Thanks！

**Comment by zxDeepDiver** [comment]
Hi, we find that it is caused by the loss weighting factor. It is not stable across different pytorch versions.  We have changed it to 1 and reruned the code. [https://github.com/CASIA-IVA-Lab/TDKD/commit/f731fb50ab1d06bb654ef030d201bad7d56ab3da ](https://github.com/CASIA-IVA-Lab/TDKD/commit/f731fb50ab1d06bb654ef030d201bad7d56ab3da )
Now it runs OK. The new result is : 

Easy   Val AP: 0.8945238697636165
Medium Val AP: 0.8781667031622746
Hard   Val AP: 0.8137315344873659

If you have any other questions, you can put it here or send email to xu.zhao@nlpr.ia.ac.cn.

**Comment by yarkable** [comment]
Many thanks to you all. @zxDeepDiver @icekrn
